### PR TITLE
Generate OpenAPI JSON schema for Terraform Component

### LINKED ISSUE #1736: Cannot store capability alibaba-rds in ConfigMap: failed to generate OpenAPI v3 JSON schema for capability alibaba-rds: capability alibaba-rds doesn't contain section `parameter`
**Describe the bug**

I tried to follow the steps to create a Terraform component at this link (https://kubevela.io/docs/platform-engineers/terraform)
but I have the error during reconciliation

**To Reproduce**
Follow this guide (https://kubevela.io/docs/platform-engineers/terraform) and try to describe the alibaba-rds component

```
k describe comp alibaba-rds -n vela-system    

Name:         alibaba-rds
Namespace:    vela-system
Labels:       <none>
Annotations:  definition.oam.dev/description: Terraform configuration for Alibaba Cloud RDS object
              type: terraform
API Version:  core.oam.dev/v1beta1
Kind:         ComponentDefinition
Metadata:
  Creation Timestamp:  2021-05-31T16:54:15Z
  Generation:          1
  Managed Fields:
    API Version:  core.oam.dev/v1alpha2
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .:
          f:definition.oam.dev/description:
          f:kubectl.kubernetes.io/last-applied-configuration:
          f:type:
      f:spec:
        .:
        f:schematic:
          .:
          f:terraform:
            .:
            f:configuration:
            f:type:
        f:workload:
          .:
          f:definition:
            .:
            f:apiVersion:
            f:kind:
    Manager:      kubectl-client-side-apply
    Operation:    Update
    Time:         2021-05-31T16:54:15Z
    API Version:  core.oam.dev/v1beta1
    Fields Type:  FieldsV1
    fieldsV1:
      f:status:
        .:
        f:conditions:
    Manager:         6b6e1c60-3398-4205-b124-4b5d6d1277a4
    Operation:       Update
    Time:            2021-05-31T16:54:18Z
  Resource Version:  4665
  Self Link:         /apis/core.oam.dev/v1beta1/namespaces/vela-system/componentdefinitions/alibaba-rds
  UID:               6b6e1c60-3398-4205-b124-4b5d6d1277a4
Spec:
  Schematic:
    Terraform:
      Configuration:  module "rds" {
  source = "terraform-alicloud-modules/rds/alicloud"
  engine = "MySQL"
  engine_version = "8.0"
  instance_type = "rds.mysql.c1.large"
  instance_storage = "20"
  instance_name = var.instance_name
  account_name = var.account_name
  password = var.password
}

output "DB_NAME" {
  value = module.rds.this_db_instance_name
}
output "DB_USER" {
  value = module.rds.this_db_database_account
}
output "DB_PORT" {
  value = module.rds.this_db_instance_port
}
output "DB_HOST" {
  value = module.rds.this_db_instance_connection_string
}
output "DB_PASSWORD" {
  value = module.rds.this_db_instance_port
}

variable "instance_name" {
  description = "RDS instance name"
  type = string
  default = "poc"
}

variable "account_name" {
  description = "RDS instance user account name"
  type = "string"
  default = "oam"
}

variable "password" {
  description = "RDS instance account password"
  type = "string"
  default = "Xyfff83jfewGGfaked"
}

      Type:  hcl
  Workload:
    Definition:
      API Version:  terraform.core.oam.dev/v1beta1
      Kind:         Configuration
Status:
  Conditions:
    Last Transition Time:  2021-05-31T16:54:18Z
    Message:               cannot store capability alibaba-rds in ConfigMap: failed to generate OpenAPI v3 JSON schema for capability alibaba-rds: capability alibaba-rds doesn't contain section `parameter`
    Reason:                ReconcileError
    Status:                False
    Type:                  Synced
```

This is the log of controller:
```
I0531 16:54:15.419218       1 componentdefinition_controller.go:62] "Reconciling ComponentDefinition" Name="alibaba-rds" Namespace="vela-system"
I0531 16:54:18.059494       1 componentdefinition_controller.go:128] "Successfully create WorkloadDefinition" name="alibaba-rds"
E0531 16:54:18.059687       1 componentdefinition_controller.go:147] "cannot store capability in ConfigMap" err="failed to generate OpenAPI v3 JSON schema for capability alibaba-rds: capability alibaba-rds doesn't contain section `parameter`"
I0531 16:54:18.070754       1 componentdefinition_controller.go:62] "Reconciling ComponentDefinition" Name="alibaba-rds" Namespace="vela-system"
I0531 16:54:18.070860       1 componentdefinition_controller.go:128] "Successfully create WorkloadDefinition" name="alibaba-rds"
E0531 16:54:18.070901       1 componentdefinition_controller.go:147] "cannot store capability in ConfigMap" err="failed to generate OpenAPI v3 JSON schema for capability alibaba-rds: capability alibaba-rds doesn't contain section `parameter`"
```

**Expected behavior**
This shold be created without any problem

**Cluster information**
```
k version
                                                                                                                                                                                                                                                                   
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.7", GitCommit:"1dd5338295409edcfff11505e7bb246f0d325d15", GitTreeState:"clean", BuildDate:"2021-01-13T13:23:52Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.15", GitCommit:"73dd5c840662bb066a146d0871216333181f4b64", GitTreeState:"clean", BuildDate:"2021-01-22T22:45:59Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

### PR DESCRIPTION
**Pull request by zzxwill**
Fix #1736

### CODE REVIEW

File: pkg/controller/utils/capability.go [hunk 1]
```diff
@@ -74,13 +86,19 @@ func NewCapabilityComponentDef(componentDefinition *v1beta1.ComponentDefinition)
...
+		if componentDefinition.Spec.Schematic.HELM != nil {
+			def.WorkloadType = util.HELMDef
+			def.Helm = componentDefinition.Spec.Schematic.HELM
+		}
+		if componentDefinition.Spec.Schematic.KUBE != nil {
+			def.WorkloadType = util.KubeDef
+			def.Kube = componentDefinition.Spec.Schematic.KUBE
+		}
+		if componentDefinition.Spec.Schematic.Terraform != nil {
+			def.WorkloadType = util.TerraformDef
```
Comment (zzxwill) [inline pkg/controller/utils/capability.go:99]: In fact, L99-L100 is covered.
image

File: pkg/controller/utils/capability.go [hunk 2]
```diff
@@ -95,6 +113,51 @@ func (def *CapabilityComponentDefinition) GetOpenAPISchema(pd *definition.Packag
 	return getOpenAPISchema(capability, pd)
 }
 
+// GetOpenAPISchemaFromTerraformComponentDefinition gets OpenAPI v3 schema by WorkloadDefinition name
+func GetOpenAPISchemaFromTerraformComponentDefinition(configuration string) ([]byte, error) {
```
Comment (zzxwill) [inline pkg/controller/utils/capability.go:117]: Covered.
image

**Comment by zzxwill** [comment]
Added one more test case, the coverage dropped from 0.12% to 4.34%.

**Comment by yangsoon** [comment]
LGTM

**Comment by hongchaodeng** [comment]
Well.. This title is confusing.. OpenAPI and JSON schema are two different types..

**Comment by zzxwill** [comment]
> Well.. This title is confusing.. OpenAPI and JSON schema are two different types..

https://swagger.io/docs/specification/data-models/
>OpenAPI 3.0 data types are based on an extended subset JSON Schema Specification Wright Draft 00 (aka Draft 5)

**Review by wonderflow — APPROVED** [review]
lgtm  please cherry-pick to release-1.0 branch

**Pull request merged**